Task3.c

```diff
@@ -0,0 +1,134 @@
+#include <stdio.h>
+#include <limits.h>
+#include <stdlib.h>
+
+
+void bubbleSort (int array[], int length)
+{
+    for (int i = 0; i < length - 1; i++)
+    {
+        for (int j = 0; j < length - i - 1; j++)
+        {
+            if (array[j] > array[j+1])
+            {
+                int temp = array[j];
+                array[j] = array[j + 1];
+                array[j + 1] = temp;
+            }
+        }
+    }
+}
+
+void printArray (int array[], int length)
+{
+    for (int i = 0; i < length; i++)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+int searchMaxInArray (int array[], int length)
+{
+    int maximum = INT_MIN;
+    for (int i = 0; i < length; i++)
+    {
+        if (array[i] > maximum)
+        {
+            maximum = array[i]; 
+        }    
+    }
+    return maximum;
+}
+
+int searchMinInArray (int array[], int length)
+{
+    int minimum = INT_MAX;
+    for (int i = 0; i < length; i++)
+    {
+        if (array[i] < minimum)
+            minimum = array[i]; 
+    }
+    return minimum;
+}
+
+int countingSort (int array[], int length)
+{
+    int maximumElement = searchMaxInArray(array, length);
+    int minimumElement = searchMinInArray(array, length);
+    int numberOfCounters = maximumElement - minimumElement + 1;
+    int* arrayOfCounters = calloc(numberOfCounters, sizeof(int));
+    if (NULL == arrayOfCounters)
+    {
+        return 0;
+    }
+    for (int i = 0; i < length; i++)
+    {
+        arrayOfCounters[ array[i] - minimumElement ]++;
```

Comment:
```suggestion
        arrayOfCounters[array[i] - minimumElement]++;
```
Вырезка из массива без пробелов